src/core/symbology/qgssymbol.cpp

```diff
@@ -1067,15 +1074,22 @@ QImage QgsSymbol::bigSymbolPreviewImage( QgsExpressionContext *expressionContext
   if ( mType == Qgis::SymbolType::Marker && flags & Qgis::SymbolPreviewFlag::FlagIncludeCrosshairsForMarkerSymbols )
   {
     p.setPen( QPen( Qt::gray ) );
-    p.drawLine( 0, 50, 100, 50 );
-    p.drawLine( 50, 0, 50, 100 );
+    p.drawLine( QLineF( 0, 50, 100, 50 ) );
+    p.drawLine( QLineF( 50, 0, 50, 100 ) );
   }
 
   QgsRenderContext context = QgsRenderContext::fromQPainter( &p );
   context.setFlag( Qgis::RenderContextFlag::RenderSymbolPreview );
   context.setFlag( Qgis::RenderContextFlag::Antialiasing );
   context.setFlag( Qgis::RenderContextFlag::HighQualityImageTransforms );
   context.setPainterFlagsUsingContext( &p );
+  context.setDevicePixelRatio( devicePixelRatio );
```

Comment:
Is that needed, you'll end up setting it a few lines down in updateRenderContextForScreen()